uwuno/src/main/java/com/learning/uwuno/controllers/playerController.java

```diff
@@ -52,40 +36,82 @@ public String addPlayer(@RequestBody String json, @PathVariable String uid)
-    public void updatePlayerName(@RequestBody String json, @PathVariable String uid)
-            throws JsonMappingException, JsonProcessingException {
+    // Possible calls for player PUT, for differing json inputs check JSON Documentation.txt
+    // 1) Update player name
+    // 2) Play Card
+    // 3) Draw Card
+    @PutMapping(value = "rooms/{uid}/players/{pid}")
+    public Object handlePlayerPuts(@RequestBody String json, @PathVariable String uid, @PathVariable String pid) {
+        parser parser;
         try {
-            // OPTIONAL TODO: Check if name already exists and append number to it
-            parser parser = new parser(json);
-            room room = containerService.getRoom(uid);
-            player player = room.getPlayer(parser.getValue("id"));
-            player.setName(parser.getValue("name"));
-        }
-        catch (NoSuchElementException e) {
-            throw new errorNotFound();
+            parser = new parser(json);
         }
         catch (JsonProcessingException e) {
             throw new internalServerError();
         }
+
+        // TODO: Change this to return a JSON Object, returning a POJO screws with JSON return
+        // Handle updating name
+        if (parser.exists("name")) {
+            try { // TODO: Check if name already exists and append number to it
+                containerService.updatePlayerName(uid, pid, parser.getValue("name"));
+                // Should we do it this way? Def good for debugging front end at least
+                return containerService.getPlayer(uid, pid);
+            }
+            catch (NoSuchElementException e) {
+                throw new errorNotFound();
+            }
+        }
+        // Handle drawing a card - remove a card from the deck and add to a player's cardList
+        else if (parser.exists("draw")) {
+            try {
+                int numToDraw = Integer.parseInt(parser.getValue("draw"));
+                return containerService.drawCards(uid, pid, numToDraw);
+            }
+            catch (NumberFormatException e) {
+                throw new badRequest();
+            }
+        }
+        // Handle playing a card - remove card from player and add to deck
+        // Will check for whether card is playable, however will throw if not
+        else if (parser.exists("cardType") &&
+                parser.exists("cardColor") &&
+                parser.exists("cardValue") &&
+                parser.exists("wildColor")) {
+            containerService.playCard(uid, pid, parser.getValue("cardType"),
+                                                parser.getValue("cardColor"),
+                                                parser.getValue("cardValue"),
+                                                parser.getValue("wildColor"));
+            return containerService.getPlayer(uid, pid).getCardList();
+        }
+        // Handle missing card information for playing
+        else if (parser.exists("cardType") ||
```

Comment:
Either delete and just do a "else throw badrequest" or change the comment to a TODO so we don't forget to create the custom message